src/subdomains/supporting/support-issue/services/support-issue.service.ts

```diff
@@ -52,36 +68,44 @@ export class SupportIssueService {
         reason: newIssue.reason,
         transaction: { id: newIssue.transaction?.id ?? IsNull() },
       },
-      relations: { messages: true },
+      relations: { messages: true, transaction: true, limitRequest: true },
     });
 
-    if (!existingIssue && dto.transaction) {
-      if (dto.transaction.id) {
-        newIssue.transaction = await this.transactionService.getTransactionById(dto.transaction.id, {
-          user: { userData: true },
-        });
-        if (!newIssue.transaction) throw new NotFoundException('Transaction not found');
-        if (!newIssue.transaction.user || newIssue.transaction.user.userData.id !== newIssue.userData.id)
-          throw new ForbiddenException('You can only create support issue for your own transaction');
+    if (!existingIssue) {
+      // create UID
+      const hash = Util.createHash(newIssue.type + new Date() + Util.randomId()).toUpperCase();
+      newIssue.uid = `${this.UID_PREFIX}${hash.slice(0, 16)}`;
+
+      // map transaction
+      if (dto.transaction) {
+        if (dto.transaction.id) {
```

Comment:
In which case is `dto.transaction` set but `dto.transaction.id` isn't set? 🤔

Reply:
It's optional 👍